### ISSUE TITLE
Unknown commands cause errors

### ISSUE THREAD

**Issue by sharang-d** [mozilla/web-ext#215]
Using any unknown command on the CLI causes a Node error.
Eg:
```
➜  bin git:(master) ✗ ./web-ext unknown

module.js:341
    throw err;
    ^

Error: Cannot find module '../dist/web-ext'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/Users/sharang/Data/open-source/mozilla/web-ext/bin/web-ext:5:1)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
```

We should probably not do that and show help text to the user instead.

**Comment by sharang-d** [comment]
@kumar303 
I think that we can handle this using the [API](https://github.com/yargs/yargs#strict) provided by yargs.
Following would be the diff:
```diff
diff --git a/src/program.js b/src/program.js
index c122628..55902fc 100644
--- a/src/program.js
+++ b/src/program.js
@@ -24,6 +24,7 @@ export class Program {
       yargsInstance = yargs(argv);
     }
     this.yargs = yargsInstance;
+    this.yargs.strict();
     this.commands = {};
   }
```

and then..
```
➜  bin git:(master) ✗ ./web-ext unknown       
Usage: web-ext [options] command

Option values can also be set by declaring an environment variable prefixed
with $WEB_EXT_. For example: $WEB_EXT_SOURCE_DIR=/path is the same as
--source-dir=/path.

To view specific help for any given command, add the command name.
Example: web-ext --help run.


Commands:
  build  Create a web extension package from source
  sign   Sign the web extension so it can be installed in Firefox
  run    Run the web extension

Options:
  --version            Show version number                             [boolean]
  --source-dir, -s     Web extension source directory.
                                                   [string] [required] [default:
                          "/Users/sharang/Data/open-source/mozilla/web-ext/bin"]
  --artifacts-dir, -a  Directory where artifacts will be saved.
                                                   [string] [required] [default:
        "/Users/sharang/Data/open-source/mozilla/web-ext/bin/web-ext-artifacts"]
  --verbose, -v        Show verbose output                             [boolean]
  -h, --help           Show help                                       [boolean]

Unknown argument: unknown
```

I can take this up if I'm on the right track here or you can tell me if there's something else missing that needs to be done to go with this.

Thanks

**Comment by kumar303** [comment]
Hmm, I think maybe `./web-ext` (executed as a local script) is just broken. Does it work for you at all? Does this work? `./web-ext --help` ?

It's not so useful to run `web-ext` like this because the current working directory always matters. For local development, the documentation instead suggests to [install it using npm link](https://github.com/mozilla/web-ext/blob/master/README.md#installation-from-source). With that method you can change into any directory.

However, I think `yargs.strict()` is the fix we need for https://github.com/mozilla/web-ext/issues/169 ! Can you try to patch it for that issue?

**Comment by sharang-d** [comment]
> Hmm, I think maybe ./web-ext (executed as a local script) is just broken. Does it work for you at all? Does this work? ./web-ext --help ?

Yup, it does! Anyway, I'll stop doing that now since it is not the recommended way to do things. Thanks for educating me. 😄 

Trying it with the build/link method does the same for me.
```
➜  web-ext git:(master) web-ext --help
Usage: /usr/local/bin/web-ext [options] command

Option values can also be set by declaring an environment variable prefixed
with $WEB_EXT_. For example: $WEB_EXT_SOURCE_DIR=/path is the same as
--source-dir=/path.

To view specific help for any given command, add the command name.
Example: /usr/local/bin/web-ext --help run.


Commands:
  build  Create a web extension package from source
  sign   Sign the web extension so it can be installed in Firefox
  run    Run the web extension

Options:
  --version            Show version number                             [boolean]
  --source-dir, -s     Web extension source directory.
                                                   [string] [required] [default:
                              "/Users/sharang/Data/open-source/mozilla/web-ext"]
  --artifacts-dir, -a  Directory where artifacts will be saved.
                                                   [string] [required] [default:
            "/Users/sharang/Data/open-source/mozilla/web-ext/web-ext-artifacts"]
  --verbose, -v        Show verbose output                             [boolean]
  -h, --help           Show help                                       [boolean]
```

```
➜  web-ext git:(master) web-ext unknown

unknown: WebExtError: unknown command: unknown
    at /Users/sharang/Data/open-source/mozilla/web-ext/dist/webpack:/src/program.js:64:17
    at lib$es6$promise$$internal$$initializePromise (/Users/sharang/Data/open-source/mozilla/web-ext/node_modules/es6-promise/dist/es6-promise.js:408:9)
    at new lib$es6$promise$promise$$Promise (/Users/sharang/Data/open-source/mozilla/web-ext/node_modules/es6-promise/dist/es6-promise.js:581:60)
    at Program.run (/Users/sharang/Data/open-source/mozilla/web-ext/dist/webpack:/src/program.js:57:12)
    at Object.main (/Users/sharang/Data/open-source/mozilla/web-ext/dist/webpack:/src/program.js:189:18)
    at Object.<anonymous> (/Users/sharang/Data/open-source/mozilla/web-ext/bin/web-ext:5:28)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:146:18)
    at node.js:404:3
```

> However, I think yargs.strict() is the fix we need for #169 ! Can you try to patch it for that issue?

That is actually what I was aiming to fix but could not and stumbled onto this. 😛 
I'm afraid that the fix for that is something similar but not this patch exactly. I think that the command and sub-command need to be handled separately and it seems that yargs expects that as well.

**Comment by kumar303** [comment]
ok, if you want to make a pull request I'll take a look
